pkg/apply/apply.go

```diff
@@ -122,19 +123,19 @@ func NewTopicApplier(
 //     c. Check partition count and extend if needed
 //     d. Check partition placement and update/migrate if needed
 //     e. Check partition leaders and update if needed
-func (t *TopicApplier) Apply(ctx context.Context) error {
+func (t *TopicApplier) Apply(ctx context.Context) (map[string]interface{}, error) {
```

Comment:
I don't think returning a free form map for the apply method would be a good idea.
Who uses this return value has no idea of what the format should be, that is very fragile.

I think you have two options:
- [my preferred] actually build a structure which represents a typed abstract data type that returns all the details of the apply operations: which field where changed, to which value, etc. Then format that into json in the callsite. Now the return type is not coupled with presentation anymore and can be used for something useful.
- Keep it returning Null, and create a separate mutable abstraction to collect the content of the output response. basically like a file, you periodically write on the logger. At the end the logger is flushed outside of this method.

Reply:
I've added a `ChangesTracker` struct to store changes made to a topic: https://github.com/getsentry/topicctl/pull/3/files#diff-e93827340ebe6b239ff60427dba9f2881d96e7bba9a8a9b07a09cb64582dd73fR95